### ISSUE TITLE
ppc64le: ".toc" entry being a relocation to ".data*" section

### ISSUE THREAD

**Issue by julien-thierry** [dynup/kpatch#1073]
Hi,

Kpatch-build does not support modificaitons to data sections and prevents such section to be included in the patch object.

One issue appears on ppc64le when the ".toc" section has a reference to a data section. ".toc" section and its relocations are always included in the final patch object as a "standard_element", and whenever a relocation is included in the patch object, the symbol/section it references get included as well. Which can cause a data section to be dragged in with the following message:

`data section .data selected for inclusion`

In the attached reproducer, neither the data nor the toc gets modified.

[data-inclusion-reprod.tar.gz](https://github.com/dynup/kpatch/files/4243756/data-inclusion-reprod.tar.gz)

**Comment by julien-thierry** [comment]
@kamalesh-babulal
Have you ever had to deal with this kind of issues? Would you have any insights on how we could safely deal with them?

Thanks

**Comment by kamalesh-babulal** [comment]
@julien-thierry, there is a long-standing issue similar to this #795, the root cause looks similar. It needs to be fixed. Following one-liner, helped in reproducing it:
```
--- a/drivers/ata/pata_piccolo.c
+++ b/drivers/ata/pata_piccolo.c
@@ -94,6 +94,8 @@ static int ata_tosh_init_one(struct pci_dev *dev, const struct pci_device_id *id
        };
        const struct ata_port_info *ppi[] = { &info, &ata_dummy_port_info };
        /* Just one port for the moment */
+       if (!jiffies)
+               printk("Kpatch\n");
        return ata_pci_bmdma_init_one(dev, ppi, &tosh_sht, NULL, 0);
 }
```
will try and debug more and update my findings/ideas.

**Comment by julien-thierry** [comment]
Yes, my reproducer actually comes from pata_piccolo.c as well. I think you don't even need to introduce a code change. Just using create-diff-object on pata_piccolo.o as orig and patched might trigger the issue (or just doing kpatch-build with a comment changed).

But yes, the root cause of the issue you pointed at seems to be the same as this one.
